src/include/daos_security.h

```diff
@@ -625,6 +626,47 @@ daos_acl_principal_from_str(const char *principal_str,
 int
 daos_acl_to_stream(FILE *stream, struct daos_acl *acl, bool verbose);
 
+/**
+ * Fetch a user's permissions for a specific pool.
+ *
+ * \param[in]	pool_prop	Pool property containing DAOS_PROP_PO_ACL/OWNER/OWNER_GROUP entries
+ * \param[in]	uid		User's local uid
+ * \param[in]	gid		Gid of the user's primary group
+ * \param[in]	supp_gids	Gids of the user's supplementary groups, if any
+ * \param[in]	nr_supp_gids	Length of the supp_gids list
+ * \param[out]	perms		Bitmap representing the user's permissions. Bits are defined
+ *				in enum daos_acl_perm.
+ *
+ * \return	0		Success
+ *		-DER_INVAL	Invalid input
+ *		-DER_NONEXIST	UID or GID not found on the system
+ *		-DER_NOMEM	Could not allocate memory
+ */
+int
+daos_sec_get_pool_permissions(daos_prop_t *pool_prop, uid_t uid, gid_t gid, gid_t *supp_gids,
```

Comment:
would it be better to make those functions in the pool and container headers and rename them to something more user-friendly?
so daos_pool_get_perm() and daos_cont_get_perm()?

Reply:
Will do.